server.cjs

```diff
@@ -0,0 +1,51 @@
+const { createServer, ServerResponse, IncomingMessage } = require('http')
+const { parse } = require('url')
+const next = require('next')
+ 
+const dev = process.env.NODE_ENV !== 'production'
+const hostname = 'localhost'
+const port = 3000
+// when using middleware `hostname` and `port` must be provided below
+const app = next({ dev, hostname, port })
+const handle = app.getRequestHandler()
+
+app.prepare().then(() => {
+  createServer(async (req, res) => {
+    // Be sure to pass `true` as the second argument to `url.parse`.
+    // This tells it to parse the query portion of the URL.
+    const parsedUrl = parse(req.url, true)
+
+    const setHeader = res.setHeader
+    let cacheControlOverwritten = false
+    res.setHeader = function(header, value) {
+      if (header.toLowerCase() === 'x-cache-control') {
+        cacheControlOverwritten = true
+        return setHeader.call(this, 'Cache-Control', value)
+      }
+
+      if (header.toLowerCase() === 'cache-control') {
+        if (cacheControlOverwritten) {
+          return this
```

Comment:
you sure? what should setHeader return in general?

Reply:
It should return `res`/`this`

<img width="742" alt="image" src="https://github.com/lidofinance/trp-ui/assets/103929444/b4657a7d-b30a-48d8-82e2-41482ed38828">

Reply:
OK, thanks